### DISCUSSION TITLE
Favorite routes

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#35] [Q&A] [answered]
In Munich, I have recently been banned from driving my car into the city center. By the way, I understand the objective - but not the execution. I would have nothing against a city toll, but to exchange my car for a new one is really not ecological!
Now I have defined a few routes to reach some destinations (my son, the library, the favorite cinema) via detours. It occurred to me that **such repeatedly used routes might also be favorites - is that a valid idea?**

**Comment by devemux86** [comment]
Thanks for the request, it is a perfectly valid idea!

Favorites are now being redesigned to introduce search and other features in the future.

We could also have favorite routes / tracks internally in app and display / manage them.
Currently we store routes as offline route files in the device and import them each time.

---

Note: route and track are different concepts:
- Track is like a recorded GPS path, only coordinates, needs route calculation to drive it.
- Route is a full navigable path, with directions etc., only needs offline loading to drive it.
Like the kurviger files or the GPX navigation track files.

Which one do you have in mind?

**Reply by WolfgangZ42** [reply]
Admittedly, I have not thought about it. Both are possible without any problems. How about allowing both types?

**Reply by devemux86** [reply]
> How about allowing both types?

We can also do this. :slightly_smiling_face: 

Cruiser already remembers the last route calculated or imported.
And it already remembers the imported tracks (with their colors).

**Comment by WolfgangZ42** [comment]
Because I did some tests yesterday, I now think that Route (navigation gpx) is best suited for favorites. This can then be imported without further details - unlike Tracks:
![2023-02-17 10 04 52](https://user-images.githubusercontent.com/121806238/219690803-6692b365-e9b1-44ed-ac4c-dfd72dd148f5.png)   ![2023-02-17 10 06 14](https://user-images.githubusercontent.com/121806238/219690854-39943a06-6b49-4046-a561-7c9e447d85a6.png)

**Reply by devemux86** [reply]
GPX navigation track is an **offline** GPX format that stores everything: route path, navigation instructions, etc.
It is used by BRouter, Locus, [RouteYou](https://www.routeyou.com/), ...

Kurviger is a similar format I wrote for K1 / K2 so can load **offline** the routes.
Then we offered it in the app and website export / import dialogs.

---

Cruiser can export and import **offline** GPX navigation tracks.

GPX navigation track can also store all the Kurviger routing parameters,
so you can convert from one format to another while keeping all options.

_And everything is done **offline**._

**Comment by devemux86** [comment]
Do we also need categories (i.e. groups / folders) for the favorite routes and tracks?

**Reply by WolfgangZ42** [reply]
Not really for my opinion - it's just a shortcut for few repeatly used routes. 
Maybe we could separate it from "normal" favorites: /favorites/list and /favorites/routes ...

**Reply by devemux86** [reply]
We could have 3 favorite types:
- Points: https://github.com/devemux86/cruiser/discussions/45
- Routes: loaded **offline** and ready for navigation
- Tracks: like GPX trk overlays?

Favorite routes should be useful?
Do we also need favorite tracks?

**Reply by WolfgangZ42** [reply]
As I said before, routes should be sufficient IMHO. 
This repeatly used route are fix and shouldn't be changed by further parameters while importing.

**Comment by DolomitesTom** [comment]
> Favorite routes should be useful?
For me yes as favorite, named route sections.

Tracks? I normally don't use tracks, maybe for others?

**Reply by devemux86** [reply]
My thinking is that Cruiser now remembers imported tracks (and colors), with no way to manage them.
Like show / hide some of them, change color, etc. Now you need to clear overlays and re-import them.

**Comment by devemux86** [comment]
For the favorite tracks, let's continue the discussion in a separate topic:
- https://github.com/devemux86/cruiser/discussions/81

**Comment by devemux86** [comment]
"**Favorite routes**" preview:

- **Offline** routes ready for navigation
- Organized in groups
- Available in search
- Import / export

<table>
<tr>
<th>Route groups</th><th>Favorite routes</th>
</tr>
<tr>
<td><img src="https://github.com/devemux86/cruiser/assets/3484020/786e69e0-ee16-45d0-861e-48c054513afa"/></td><td><img src="https://github.com/devemux86/cruiser/assets/3484020/ed3e9343-0995-4b3a-9722-5875a96dba17"/></td>
</tr>
<tr>
<th>Favorites menu</th><th>Search routes</th>
</tr>
<tr>
<td><img src="https://github.com/devemux86/cruiser/assets/3484020/4abefdb8-f07f-4bae-9b28-a9344ff04d66"/></td><td><img src="https://github.com/devemux86/cruiser/assets/3484020/e33cb75e-9371-4687-b6be-ebe4974731d3"/></td>
</tr>
</table>

**Reply by Ciutnik** [reply]
One note related to naming: In the third picture (Favorites menu), under the star, there should probably be the word "Points"? This would be consistent with "Routes" and "Tracks".

**Reply by devemux86** [reply]
@Ciutnik good point! :slightly_smiling_face:

**Reply by bios4** [reply]
@devemux86 My suggestion would be to name it "Places" and not "Points". What do you think?

**Reply by devemux86** [reply]
@bios4 We could also use this.

**Reply by Ciutnik** [reply]
Yes, since we will have favorite not only Points, but Routes and Tracks, the term "Places" seems good. But it will require replacing the existing "Favorites" (as points) throughout the application.

**Reply by devemux86** [reply]
This is difficult, also "favorites" is a more common term vs places (or points),
about the actual meaning of what "favorite" or "bookmark" actually represents.

But we can use the "places" at least in app screens where all types appear together.

**Reply by devemux86** [reply]
For example, long press on a route on the map, we can set a favorite (like now) or add the route in favorite routes, with a menu:
(the term "favorite" is needed here)
- Favorite
- Favorite route

**Reply by bios4** [reply]
I didn't mean to rename the whole "Favorites" stuff to "Places", but only the item with the "star"-symbol in above screenshot!
So all of them should be (stay) "favorites", but "single points" should be named "Places", routes be "Routes" and tracks be "Tracks".

Apologies if my suggestion has caused confusion!

**Reply by Ciutnik** [reply]
I understand the @bios4 proposal and I think it is good. "Favorites" would be a general category (or just the title of a menu window) containing:
- Places _(what was previously called "Favorites", i.e. our points)_
- Routes _(novelty)_
- Tracks _(novelty)_

If we want to add/change/remove an element to a particular group, we select one of these three. That would probably be intuitive, understandable and unambiguous, but it would require changing in several places in the application "Favorite(s)" -> "Place(s)". Also in translations!

**Reply by devemux86** [reply]
> it would require changing in several places in the application "Favorite(s)" -> "Place(s)"

This is what I would like to avoid. :slightly_smiling_face: 
Or we can keep the "Favorites" word for points and just add "Routes" + "Tracks" for the other types.

**Reply by bios4** [reply]
My vote would be to call the group "Favorites" and add "Places", "Routes" and "Tracks" to it.

> Or we can keep the "Favorites" word for points and just add "Routes" + "Tracks" for the other types.

Is also OK, but then there is the need for a name for this group of "Favorites, Routes & Tracks"...
As I understand all of these items are stored locally in the phone, so the term "My Collection" might fit quite good!??

If you think about storing all of these also online in the future, then maybe it is better to call it "Cloud".
Similar to what Kurviger is offering with K3.

**Reply by devemux86** [reply]
For example, OsmAnd has:

- My places
  - Favorites (for points)
  - Tracks

![OsmAnd](https://github.com/devemux86/cruiser/assets/3484020/99d5fbec-82d3-4de2-921d-0284cd6d0142)

**Reply by Ciutnik** [reply]
> This is what I would like to avoid.

I can imagine it and understand it. OK, these are just words 🙂

**Comment by airald** [comment]
How are stored? Can use external paths or are cruiser groups that have a list of selected routes?

**Reply by devemux86** [reply]
They are stored locally and are always available **offline**.
We can export them to known formats, as well as import them.

Or @airald do you mean something else?

**Reply by airald** [reply]
I usually store my routes and tracks in a folder on SD (and sync over the cloud). My question is if we can select the path where the favourites are stored.

**Reply by devemux86** [reply]
@airald this would be useful for all favorites: points, routes, tracks.

Although for performance reasons Cruiser uses special (internal) storage of all favorites.
Also any public folder is under the new Android restrictions (like downloaded maps / poi).

Can the /Android/media/ folders in internal storage or SD card be used in your cloud?

**Reply by airald** [reply]
I can use any folder I want, internal or SD card. I use SD card for historical reasons, I can change it easily.

**Reply by devemux86** [reply]
@airald the folders in /Android/media/ are app folders.
They may be not accessible by all external apps.

That's why I ask if your cloud app can use them?
What cloud service is it?

**Reply by airald** [reply]
I use Dropsync to sync some dropbox foders between different devices. It can access to /Android/media/gr.talent.cruiser and add and sync subfolders here.

**Reply by devemux86** [reply]
It is possible to have the profiles + favorites (points, routes, tracks) in the storage folder selected in app settings.
(together with the downloaded maps / poi)

**Reply by airald** [reply]
The sync between devices will be easier!

**Reply by devemux86** [reply]
We still have to use the import functions in Cruiser.
We can't just put random files in these folders.

**Reply by devemux86** [reply]
Sync should work:
- Import the files on one device
- Sync
- Other device gets the same (imported) files

**Reply by airald** [reply]
It's what I do, for me it's ok.

**Reply by DolomitesTom** [reply]
what is meant with sync? Do you mean export and import a profile?

**Reply by devemux86** [reply]
Sync with Google Drive, Dropbox, etc. some folders between devices.
We can move the favorites / routes / tracks in more public folders, so they can be accessed by such services.

**Reply by Ciutnik** [reply]
Or maybe move the export/import to the window header as, for example, three dots on the right? After touching, you could use the checkbox to select Favorites, Routes, Tacks for export/import? Something like that:

<img src="https://github.com/devemux86/cruiser/assets/122647782/447c8e7b-6cef-4390-b25c-f2287266b9c4" width="50%"/>

By the way, the same three dots (or six, so that it would be the same everywhere) in the header of the Profiles window would suggest the action after tapping. This would become less of a "hidden feature".

**Reply by bios4** [reply]
I strongly recommend to NOT change the current import/export function implementation! It is nicely accessible via the "six dots menu" in the favorites list:

![Screenshot_20230922_094817_Cruiser](https://github.com/devemux86/cruiser/assets/145460147/869b3a14-2c88-4149-bf83-a7a8d40266a5)

I am not a fan of such "hidden features" (like the current profile manager), I prefer when everything is immediately and directly recognizable and accessible without the need to "tap somewhere to check if there is some functionality hidden behind"...

**Reply by devemux86** [reply]
Better not move such functions to the external UI.
It's not that simple as we usually export a specific item so we need each manager to see its item lists.

---

About the profiles, lets discuss it here:
- https://github.com/devemux86/cruiser/discussions/165#discussioncomment-7079182

**Comment by devemux86** [comment]
Favorite routes can be loaded as:
- **Route**: offline and ready for navigation
- **Track**: overlay their path with any color

**Comment by devemux86** [comment]
Favorite routes are available for testing in:
- https://github.com/devemux86/cruiser/discussions/329

**Comment by devemux86** [comment]
Based on this discussion:
- https://github.com/devemux86/cruiser/discussions/335

We can also offer in the routes the same overlay functionality as in the tracks:
- Each route can have its track visible or not (add eye symbol)
- Each route can have a different track color (add color function)

**Comment by airald** [comment]
It's possible to add the current recordered track to favorites just with 2 taps. Is there the equivalent action for current route?

**Reply by devemux86** [reply]
@airald long press on the route line and click "Add favorite route".

**Reply by airald** [reply]
oh yes... simple and easy!

**Comment by devemux86** [comment]
> We can also offer in the routes the same overlay functionality as in the tracks:
> - Each route can have its track visible or not (add eye symbol)
> - Each route can have a different track color (add color function)

Preview:

<img src="https://github.com/devemux86/cruiser/assets/3484020/95152bf5-dac6-492d-a99e-4feae8d3c5ab" width="50%"/>

**Reply by devemux86** [reply]
> We can also offer in the routes the same overlay functionality as in the tracks:
> - Each route can have its track visible or not (add eye symbol)
> - Each route can have a different track color (add color function)

Available in the latest:
- https://github.com/devemux86/cruiser/discussions/329

**Comment by devemux86** [comment, marked as answer]
Favorite routes is available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
